### PR TITLE
sockpuppetbrowser: add new chart

### PR DESCRIPTION
**Pull request by robinvalk**
**Description of the change**
Add support for sockpuppetbrowser. The [docker-compose file](https://github.com/dgtlmoon/changedetection.io/blob/master/docker-compose.yml#L82) of changedetection-io recommends using it.

**Benefits**
Using this as a dependency for changedetection-io enables you to track changes on websites using screenshots etc.

**Possible drawbacks**
The sockpuppetbrowser is not released using tags on the container, we can only target "latest" 😞. Also the container needs added capabilities, in the docker-compose file linked above they just use `SYS_ADMIN` so I took that over. I don't know if there is a way to detect what the least privileges could be.

**Applicable issues**
N/A

**Additional information**

**Checklist** 
- [x] Chart version in `Chart.yaml` has been bumped according to [Semantic Versioning](https://semver.org/).
- [x] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [x] Variables have been documented in the `values.yaml` file.

_(Feel free to apply some patches to the PR yourself if that's easier for you instead of having back and fort conversation here. I'll checkout the changes you did and hopefully apply them myself correctly next time I offer to add another app.)_

### CODE REVIEW

**Comment by robinvalk** [comment]
I think I miss a docs/....md file. Are they automatically generated somehow or should I copy/paste an existing one and update its contents?

**Comment by robinvalk** [comment]
Can we somehow make this a dependency of the changedetection-io chart and automatically configure it to use this browser? In the changedetection docker-compose file it's clear how that should be done on the application level.

Now I wonder how do you have changedetection currently running without this chart? 😅

**Comment by robinvalk** [comment]
> I think I miss a docs/....md file. Are they automatically generated somehow or should I copy/paste an existing one and update its contents?

Also I am not sure what to do with the Readme.md of the chart, is it generated?

File: charts/apps/sockpuppetbrowser/values.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    enabled: false
+
+persistence: {}
+
+resources:
+  requests:
+    cpu: 100m
+    memory: 128Mi
+  limits:
+    memory: 512Mi
```
Comment (zekker6) [inline charts/apps/sockpuppetbrowser/values.yaml:34]: It is usually preferred for chart to not specify resource requests limits and let the user to set limits if needed.

**Comment by zekker6** [comment]
> I think I miss a docs/....md file. Are they automatically generated somehow or should I copy/paste an existing one and update its contents?

It is generated and copied via scripts later on. 
I'm using a [Taskfile](https://taskfile.dev/) configuration located in the repository root [here](https://github.com/zekker6/helm-charts/blob/main/Taskfile.yaml) in order to run these commands locally. 
For local development you can use `task precommit:all` and it will generate docs and do linting of the changes.


> Can we somehow make this a dependency of the changedetection-io chart and automatically configure it to use this browser? In the changedetection docker-compose file it's clear how that should be done on the application level.

Yes, once this chart will be published we will be able to add it as a dependency to an existing changedetection chart.

> Now I wonder how do you have changedetection currently running without this chart? 😅

In my use-cases I'm tracking a few websites which do not require client-side rendering, so checking for HTML-based diff works very well.

> Also I am not sure what to do with the Readme.md of the chart, is it generated?

Yes, it is generated from `README_CONFIG.md.gotmpl` file in the chart root. You can check some other readme contents for an example - https://github.com/zekker6/helm-charts/blob/main/charts/apps/homer/README_CONFIG.md.gotmpl

**Comment by robinvalk** [comment]
I've processed the feedback. Let me know if I missed something!

**Review by zekker6 — APPROVED** [review]
LGTM, @robinvalk thank you for your contribution!

**Pull request merged**